pkg/infra/grn/grn.go

```diff
@@ -21,6 +20,28 @@ type GRN struct {
 	// ResourceIdentifier is used by the underlying service to identify the
 	// resource.
 	ResourceIdentifier string
+
+	// GRN can not be extended
+	_ interface{}
+}
+
+// MarshalJSON marshals GRN as a string in JSON.
+func (g GRN) MarshalJSON() ([]byte, error) {
+	v := g.String()
+
+	return []byte(`"` + v + `"`), nil
+}
+
+// UnmarshalJSON reads a json value into GRN
+func (g *GRN) UnmarshalJSON(data []byte) error {
```

Comment:
It was useful in my first pass where I stuck GRN (as an object) on the test object -- but later replaced it with the string version.

Thoughts?

Reply:
That's interesting. Is there a reason to favor this instead of adding `json` tags to the GRN struct and using the standard json unmarshal method instead?

Reply:
It is a good question -- and brings us back a bit to when the string for struct forms make the most sense.  When and where do we want to pass things as a string grn vs an explicitly typed one.

I will make a new PR/branch with a protobuf typed struct for resource id -- and then we can evaluate